drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -45,6 +47,13 @@ static int i2c_nrfx_twim_transfer(const struct device *dev,
 		.address = addr
 	};
 
+	/* If for whatever reason the TWIM peripheral is still not initialized
+	 * at this point, try to initialize it now.
+	 */
+	if (!dev_data->twim_initialized && init_twim(dev) < 0) {
+		return -EIO;
+	}
```

Comment:
what could be the reason? If it's because of PM, consider using runtime PM API.

Reply:
Not only PM. Reinitialization of the nrfx driver could also potentially fail when `i2c_configure()` was called. One reason for this that I can imagine is that one tried to configure the 1 MHz frequency on nRF5340 for TWIM using other pins that those dedicated ones (p1.02, p1.03), on which such frequency is only available. In such case, the above call to `init_twim(dev)` would anyway fail also here, so I could just use:
```
	if (!dev_data->twim_initialized) {
		return -EIO;
	}
```
but I thought I'd rather try to reinitialize TWIM here, just in case the reason for the previous fail of `init_twim(dev)` was different (something I could not anticipate now).